crates/core/src/util/validate_addr.rs

```diff
@@ -0,0 +1,32 @@
+use std::str::FromStr;
+
+/// Trait containing functions to validate Blockchain address
+pub trait ValidateAddress {
+    /// Checks if it is an EVM address
+    fn is_evm_address(self) -> bool;
+
+    #[cfg(feature = "solana")]
+    /// Checks if it is a Solana address
+    fn is_solana_address(self) -> bool;
+}
```

Comment:
nice. now we cooking. good to have method that checks both `fn is_blockchain_address(self) -> bool;`